assume/units/dst_components.py

```diff
@@ -2,782 +2,1617 @@
+                - `cop_constraint[t]`: Enforces the relationship between power input and heat output based on the coefficient of performance (COP).
+                - `operating_cost_constraint[t]`: Calculates the operating cost based on the power input and electricity price.
+                - `ramp_up_constraint[t]`: Limits the increase in power input from one time step to the next according to the ramp-up rate.
+                - `ramp_down_constraint[t]`: Limits the decrease in power input from one time step to the next according to the ramp-down rate.
+                - `min_operating_time_constraint[t]`: Ensures the heat pump operates for at least the specified minimum number of consecutive time steps.
+                - `min_downtime_constraint[t]`: Ensures the heat pump remains off for at least the specified minimum number of consecutive time steps after shutdown.
+
+        Args:
+            model (pyo.ConcreteModel): A Pyomo ConcreteModel object representing the optimization model.
+            model_block (pyo.Block): A Pyomo Block object to which the heat pump block will be added.
+
+        Returns:
+            pyo.Block: A Pyomo block representing the heat pump with variables and constraints.
         """
-        Ensures that the power input to the electrolyser does not fall below the minimum required power.
 
-        """
-        return b.power_in[t] >= b.min_power
+        # Define parameters
+        model_block.max_power = pyo.Param(initialize=self.max_power)
+        model_block.min_power = pyo.Param(initialize=self.min_power)
+        model_block.cop = pyo.Param(initialize=self.cop)
+        model_block.ramp_up = pyo.Param(initialize=self.ramp_up)
+        model_block.ramp_down = pyo.Param(initialize=self.ramp_down)
+        model_block.min_operating_steps = pyo.Param(initialize=self.min_operating_steps)
+        model_block.min_down_steps = pyo.Param(initialize=self.min_down_steps)
+
+        # Define variables
+        model_block.power_in = pyo.Var(
+            self.time_steps, within=pyo.NonNegativeReals, bounds=(0, self.max_power)
+        )
+        model_block.heat_out = pyo.Var(self.time_steps, within=pyo.NonNegativeReals)
+        model_block.operating_cost = pyo.Var(self.time_steps, within=pyo.Reals)
+
+        # Coefficient of performance (COP) constraint
+        @model_block.Constraint(self.time_steps)
+        def cop_constraint(b, t):
+            return b.heat_out[t] == b.power_in[t] * b.cop
+
+        # Operating costs
+        @model_block.Constraint(self.time_steps)
+        def operating_cost_constraint_rule(b, t):
+            return b.operating_cost[t] == b.power_in[t] * model.electricity_price[t]
+
+        # Ramp-up constraint and ramp-down constraints
+        add_ramping_constraints(
+            model_block=model_block,
+            ramp_up=self.ramp_up,
+            ramp_down=self.ramp_down,
```

Comment:
@nick-harder just saw it now during household adaption. Is it on purpose to use the variables self.ramp_* instead of the defined pyo.Params? If not those function parameter are not needed since they are already in the block.
I also saw also other constraints all over the file, where just the variable was used.

Reply:
@HafnerMichael oh yeah true, just noticed that myself. At the beginning I didn't have the class variables, so I used the model ones. And also the idea of having the class variables is just in case we will have some new functions added to the dst classes in the future, for example to change some consraints to adapt the operating regime. And I didn't remove the model parameters. What do you think would be the better approach to avoid having duplicates? @Manish-Khanra